enarx-keep-sev/src/vm.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::x86_64::*;
+
+use std::error::Error;
+
+use kvm_bindings::kvm_userspace_memory_region;
+use kvm_ioctls::{Kvm, VmFd};
+use memmap::*;
+use x86_64::structures::paging::page_table::PageTableFlags;
+use x86_64::PhysAddr;
+
+pub struct VirtualMachine {
+    kvm: Kvm,
+    kvm_fd: VmFd,
+    address_space: Vec<kvm_userspace_memory_region>,
+    mmap_handles: Vec<MmapMut>,
+}
+
+impl VirtualMachine {
+    pub fn new(mem_size: usize) -> Result<Self, Box<dyn Error>> {
```

Comment:
We will, very soon, need to make it possible to grow and shrink the memory allocation to the VM. Can we just define an internal constant for the minimum amount of memory allocated to the VM rather than pawning off this problem to the caller who has no knowledge of the internals?